### PR TITLE
LPS-162877: Expose permissions for the object entries collections

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**

- Add GET and PUT permission endpoints in rest-openapi.yaml.
- Implement `getPermissionCheckerPortletName` method.

**NEW ENDPOINTS**
They are created for Objects which are scoped by `companyId` and `groupId`.

**CompanyId**
image

**SiteId**
image

**RESPONSE FROM A GET AND PUT REQUEST**
```
{
  "actions": {
    "get": {
      "method": "GET",
      "href": "http://localhost:8080/o/c/newscopedobjects/scopes/20122/permissions"
    },
    "replace": {
      "method": "PUT",
      "href": "http://localhost:8080/o/c/newscopedobjects/scopes/20122/permissions"
    }
  },
  "facets": [],
  "items": [
    {
      "actionIds": [
        "PERMISSIONS",
        "ADD_OBJECT_ENTRY"
      ],
      "roleName": "Owner"
    },
    {
      "actionIds": [
        "PERMISSIONS",
        "ADD_OBJECT_ENTRY"
      ],
      "roleName": "Site Member"
    }
  ],
  "lastPage": 1,
  "page": 1,
  "pageSize": 2,
  "totalCount": 2
}
```

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-162877

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 1]
```diff
@@ -273,6 +308,33 @@ aggregation, _getDTOConverterContext(null), pagination,
 			search, sorts);
 	}
 
+	public Page<Permission> getScopeScopeKeyPermissionsPage(
```
Comment (LuismiBarcos) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:311]: isn't this an Overridden method?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 2]
```diff
@@ -489,6 +637,37 @@ private void _loadObjectDefinition(Map<String, Serializable> parameters)
 		throw new NotFoundException("Missing parameter \"objectDefinitionId\"");
 	}
 
+	private Page<Permission> _toPermissionPage(
```
Comment (LuismiBarcos) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:640]: I don't get this method. Is not the same as [this one](https://github.com/liferay-headless/liferay-portal/blob/aa7f9d5bc1c18100b107f90124fdc287756aa9b2/modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/BaseObjectEntryResourceImpl.java#L1140-L1170) that it is in the Base?
Or I'm missing something?? I'm doing a quick review so maybe I'm missing something 😁

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 3]
```diff
@@ -352,6 +414,41 @@ public ObjectEntry putObjectEntry(
...
+
+		resourcePermissionLocalService.updateResourcePermissions(
+			contextCompany.getCompanyId(), siteId, portletName,
+			String.valueOf(siteId),
+			ModelPermissionsUtil.toModelPermissions(
+				contextCompany.getCompanyId(), permissions, siteId, portletName,
+				resourceActionLocalService, resourcePermissionLocalService,
+				roleLocalService));
+
+		return toPermissionPage(
```
Comment (LuismiBarcos) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:437]: Here you are calling the base method

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 4]
```diff
@@ -273,6 +308,33 @@ aggregation, _getDTOConverterContext(null), pagination,
...
+
+		Long siteId = _getGroupId(scopeKey);
+
+		String portletName = getPermissionCheckerPortletName(siteId);
+
+		PermissionUtil.checkPermission(
+			ActionKeys.PERMISSIONS, groupLocalService, portletName, siteId,
+			siteId);
+
+		return _toPermissionPage(
```
Comment (LuismiBarcos) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:323]: Here you calling the private method

**Review by LuismiBarcos — CHANGES_REQUESTED** [review]
Review my comments. Maybe I'm missing something but I think there is a method that is not necessary.

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java (hunk 2 above)
Comment (sergiojimcos) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:640]: You are right @LuismiBarcos , I didn't figure out that toPermissionPage is a protected method :sweat_smile:. I'll delete it

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Review by LuismiBarcos — COMMENTED** [review]
The code looks good to me 👏 

But I am not approving this because I tested the functionality and saw some strange behavior. 

What I did was add permissions to the custom objects for a new role and a Site Member and I did not get those permissions in the response. I always get the `Owner` permissions.

I don't know if I'm testing it right or if I am missing some steps but I want to share this information with you just in case.

This happens for both, company and site scoped custom objects.

**Comment by LuismiBarcos** [comment]
I would review the behavior of this before sending it because seems not consistent between the REST APIs and the UI

**Comment by sergiojimcos** [comment]
ci:close

**Pull request closed without merging**